src/sentry/tasks/releasemonitor.py

```diff
@@ -28,51 +28,47 @@ def monitor_release_adoption(**kwargs):
     with metrics.timer("sentry.tasks.monitor_release_adoption.aggregate_projects.loop"):
-        #     aggregated_projects = defaultdict(list)
-        #     start_time = time.time()
-        #     offset = 0
-        #     while (time.time() - start_time) < MAX_SECONDS:
-        #         query = Query(
-        #             dataset="sessions",
-        #             match=Entity("org_sessions"),
-        #             select=[
-        #                 Column("org_id"),
-        #                 Column("project_id"),
-        #             ],
-        #             groupby=[Column("org_id"),Column("project_id")],
-        #             where=[
-        #                 Condition(
-        #                     Column("started"), Op.GTE, datetime.utcnow() - timedelta(hours=6)
-        #                 ),
-        #                 Condition(Column("started"), Op.LT, datetime.utcnow()),
-        #             ],
-        #             granularity=Granularity(3600),
-        #             orderby=[OrderBy(Column("org_id"), Direction.ASC)],
-        #         ).set_limit(CHUNK_SIZE+1).set_offset(offset)
-        #         data = snuba.raw_snql_query(query, referrer="tasks.monitor_release_adoption")["data"]
-        #         count = len(data)
-        #         more_results = count >= CHUNK_SIZE
-        #         offset += count
-
-        #         for row in data:
-        #             aggregated_projects[row['org_id']].append(row['project_id'])
-
-        #         if not more_results:
-        #             break
-
-        #     else:
-        #         logger.info(
-        #             "monitor_release_adoption.loop_timeout",
-        #             extra={"offset": offset},
-        #         )
-        # NOTE: Hardcoded data for sentry org and sentry project for early release, in the same format snuba should return
-        aggregated_projects = {
-            1: [1],
-            315582: [5245995],
-            307710: [5227327],
-            34825: [1769835],
-        }  # org id: [project ids]
+        aggregated_projects = defaultdict(list)
+        start_time = time.time()
+        offset = 0
+        while (time.time() - start_time) < MAX_SECONDS:
+            query = (
+                Query(
+                    dataset="sessions",
+                    match=Entity("org_sessions"),
+                    select=[
+                        Column("org_id"),
+                        Column("project_id"),
+                    ],
+                    groupby=[Column("org_id"), Column("project_id")],
+                    where=[
+                        Condition(
+                            Column("started"), Op.GTE, datetime.utcnow() - timedelta(hours=6)
+                        ),
+                        Condition(Column("started"), Op.LT, datetime.utcnow()),
+                    ],
+                    granularity=Granularity(3600),
+                    orderby=[OrderBy(Column("org_id"), Direction.ASC)],
```

Comment:
I just noticed this, but you should order by project_id here as well, to ensure results are consistent as you paginate.

Reply:
Yeah, thanks for pointing this out. I actually noticed it too when thinking about Dan's suggestion (that I'd have to add it in, but that I should just be adding it now)